### PR TITLE
Feature/add controllers rspec

### PR DESCRIPTION
**Pull request by Fourteen98**
### Added RSpec action that checks 

- If response status was correct.
- If a correct template was rendered.
- If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -1,10 +1,41 @@
...
+    it 'returns http success on a single post' do
+      get '/users/1/posts/2'
+      expect(response).to have_http_status(:success)
+    end
+    it 'return rendered show page' do
+      get '/users/1/posts/2'
+      expect(assigns(:post)).to render_template('show')
+    end
+    it ' shows a single post' do
+      get '/users/1/posts/2'
```
Comment (Davidosky007) [inline spec/requests/posts_spec.rb]: * Kindly fix failing tests I get when I run `bundle exec rspec` 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @Fourteen98 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* No linter errors 👍 
* Good looking PR description 👍 
* Proper use of Gitflow 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- Updating your readme file setup section to suits this milestone so that visitors would find it easy installing and running your tests and project.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_spec.rb (hunk 1 above)
Comment (zainsadaqat) [inline spec/requests/posts_spec.rb]: As requested by the Previous Code Reviewer, Please fix the failed test appearing in the posts_spec.rb file 👍🏻 

Screenshot 2022-10-27 at 8 08 51 PM

File: spec/requests/users_spec.rb [hunk 2]
```diff
@@ -1,10 +1,41 @@
...
+    it 'returns http success on a single user' do
+      get '/users/1'
+      expect(response).to have_http_status(:success)
+    end
+    it 'return rendered show page' do
+      get '/users/1'
+      expect(assigns(:user)).to render_template('show')
+    end
+    it ' shows a single user' do
+      get user_path(1)
```
Comment (zainsadaqat) [inline spec/requests/users_spec.rb]: As requested by the Previous Code Reviewer, Please fix the failed test appearing in the users_spec.rb file 👍🏻


Screenshot 2022-10-27 at 8 09 40 PM

**Review by zainsadaqat — CHANGES_REQUESTED** [review]
# Required Changes 🔁 
# Hello @Fourteen98  😃 

You're doing a great job on your project. We're proud of you. Good job so far! 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
  ✅  Correct use of Github Flow
  ✅  No linter errors
  ✅  Descriptive Pull Request

## Required Changes ♻️

_Check the comments under the review._




## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding! 👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_spec.rb (hunk 1 above)
Comment (Fourteen98) [reply spec/requests/posts_spec.rb]: image

Comment (Fourteen98) [reply spec/requests/posts_spec.rb]: i think you reviewed an old commit, you can see my changes on the right

File: spec/requests/users_spec.rb (hunk 2 above)
Comment (Fourteen98) [reply spec/requests/users_spec.rb]: Screenshot 2022-10-27 at 15 20 22
I have fix all controller test errors. kindly check in my recent commit. image above show test results.

File: spec/requests/posts_spec.rb (hunk 1 above)
Comment (Fourteen98) [reply spec/requests/posts_spec.rb]: Screenshot 2022-10-27 at 15 20 22
I have fixed all controllers test errors. kindly check in my recent commit with name `fix test`

**Review by descholar-ceo — APPROVED** [review]
Hi @Fourteen98 🤝 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Implemented all of the requested changes ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**